src/api/runs.go

```diff
@@ -9,11 +12,39 @@ func handleGetJobRuns(c *gin.Context) {
 }
 
 func handleTriggerJobRun(c *gin.Context) {
-	c.String(200, "TODO")
+	jobId := c.Param("jobid")
+	namespace, name, err := helpers.SplitMetronomeJobId(jobId)
+	if err != nil {
+		JsonError(c, err.Error())
+		return
+	}
+	cronJob, err := kube.GetCronJob(namespace, name)
+	if err != nil {
+		JsonError(c, fmt.Sprintf("cannot retrieve job: %s", err))
+		return
+	}
+	job, err := kube.CreateJobFromCronjob(cronJob)
+	if err != nil {
+		JsonError(c, fmt.Sprintf("cannot run job: %s", err))
+		return
+	}
+	c.JSON(201, helpers.JobKubernetesToMetronome(job))
 }
 
 func handleGetJobRun(c *gin.Context) {
-	c.String(200, "TODO")
+	//jobId := c.Param("jobid")
+	runId := c.Param("runid")
+	namespace, name, err := helpers.SplitMetronomeJobId(runId)
+	if err != nil {
+		JsonError(c, err.Error())
+		return
+	}
+	job, err := kube.GetJob(namespace, name)
+	if err != nil {
+		JsonError(c, fmt.Sprintf("cannot get job: %s", err))
```

Comment:
metronome api specifies to throw 404 here (unless you expect it to fail due to different reasons than job not existing)

Reply:
The kube library just returns an error and doesn't really differentiate between "not found" and error, but you're probably right about it being more likely that an error here just means the job isn't found. However, this logic would need to be fixed in more than just this one place.